osmdroid-android/src/main/java/org/osmdroid/views/MapView.java

```diff
@@ -667,14 +688,66 @@ public void setScrollableAreaLimit(BoundingBoxE6 boundingBox) {
      *            A lat/long bounding box to limit scrolling to, or null to remove any scrolling
      *            limitations
      */
-    public void setScrollableAreaLimitDouble(BoundingBox boundingBox) {
-        mScrollableAreaBoundingBox = boundingBox;
-    }
+	public void setScrollableAreaLimitDouble(BoundingBox boundingBox) {
+    	if (boundingBox == null) {
+    		resetScrollableAreaLimitLatitude();
+    		resetScrollableAreaLimitLongitude();
+		} else {
+			setScrollableAreaLimitLatitude(boundingBox.getActualNorth(), boundingBox.getActualSouth(), 0);
+			setScrollableAreaLimitLongitude(boundingBox.getLonWest(), boundingBox.getLonEast(), 0);
+		}
+	}
 
+	/**
+	 * @since 6.0.0
+	 */
+	public void resetScrollableAreaLimitLatitude() {
+		mScrollableAreaLimitLatitude = false;
+	}
+
+	/**
+	 * @since 6.0.0
+	 */
+	public void resetScrollableAreaLimitLongitude() {
+		mScrollableAreaLimitLongitude = false;
+	}
+
+	/**
+	 * @since 6.0.0
+	 */
+	public void setScrollableAreaLimitLatitude(final double pNorth, final double pSouth,
+											   final int pExtraHeight) {
+		mScrollableAreaLimitLatitude = true;
+		mScrollableAreaLimitNorth = pNorth;
+		mScrollableAreaLimitSouth = pSouth;
+		mScrollableAreaLimitExtraHeight = pExtraHeight;
+	}
+
+	/**
+	 * @since 6.0.0
+	 */
+	public void setScrollableAreaLimitLongitude(final double pWest, final double pEast,
```

Comment:
is extra width distance in meters, pixels or degrees longitude?

Reply:
In pixels: currently including "pixel" in the parameter's name.

Reply:
one last question, how does the "extra" distance effect scrolling? is it like padding?

Reply:
There's no real impact on the actual scrolling, except that you can scroll further "extra" pixels beyond the scroll limits.
And to address @secaw 's needs, you can put `MapView.getWidth()/2` and `MapView.getHeight()/2` as longitude/latitude extras: doing so the map can actually be centered via `MapView.setCenter` on any `GeoPoint` within the scroll limits, which was not the case before (before, it was like we had extras of 0 pixel).

Reply:
@spyhunter99 
> is it like padding?

Yes, would actually be most descriptive!
pPaddingPixelHeight / pPaddingPixelWidth
instead of pExtra... but well, I think now it's understandable too.

Reply:
@secaw see the updated docs
https://github.com/osmdroid/osmdroid/blob/master/osmdroid-android/src/main/java/org/osmdroid/views/MapView.java#L761